d2/src/main/java/com/linkedin/d2/xds/XdsClientImpl.java

```diff
@@ -260,22 +269,64 @@ void handleD2URIMapResponse(DiscoveryResponseData data)
         {
           _log.warn("Received a D2URIMap response with no data, resource is : {}", resourceName);
         }
-        updates.put(resourceName, new D2URIMapUpdate(resource.getVersion(), nodeData));
+        updates.put(resourceName, new D2URIMapUpdate(nodeData));
       }
       catch (InvalidProtocolBufferException e)
       {
         _log.warn("Failed to unpack D2URIMap response", e);
         errors.add("Failed to unpack D2URIMap response");
         // Assume that the resource doesn't exist if it cannot be deserialized instead of simply ignoring it. This way
         // any call waiting on the response can be satisfied instead of timing out.
-        updates.put(resourceName, new D2URIMapUpdate(resource.getVersion(), null));
+        updates.put(resourceName, new D2URIMapUpdate(null));
       }
     }
     sendAckOrNack(data.getResourceType(), data.getNonce(), errors);
     handleResourceUpdate(updates, data.getResourceType());
     handleResourceRemoval(data.getRemovedResources(), data.getResourceType());
   }
 
+  @VisibleForTesting
+  void handleD2URIResponse(DiscoveryResponseData data)
+  {
+    Map<String, D2URICollectionUpdate> updates = new HashMap<>();
+    List<String> errors = new ArrayList<>();
+
+    for (Resource resource : data.getResourcesList())
+    {
+      try
+      {
+        XdsD2.D2URI uri = resource.getResource().unpack(XdsD2.D2URI.class);
+        getCollectionUpdate(updates, resource.getName())
```

Comment:
instead of passing in the new map "updates", this should be getting the existing update in current subscriber of this glob collection, so that the diffs can be accumulated to the same map to build up the full set of uris of a cluster.